### ISSUE TITLE
Chapter 13. Table13-2

### ISSUE THREAD

**Issue by fanyike** [ageron/handson-ml#202]
Should AlexNet has 227x227 input size and use valid padding in the first convolution layer?

**Comment by ageron** [comment]
Thanks for your question.  The [original paper](https://www.nvidia.cn/content/tesla/pdf/machine-learning/imagenet-classification-with-deep-convolutional-nn.pdf) mentions that the input images are 224x224, with kernel size 11x11 and stride 4. On figure 2 you can see that the resulting feature maps are 55x55.

So we have a problem: if they used VALID padding, then the resulting feature maps should be 54x54, and if they used SAME padding (at least if it is implemented like in TensorFlow), then the resulting feature maps should be 56x56.

```python
>>> import tensorflow as tf
>>> X = tf.placeholder(tf.float32, shape=[None, 224, 224, 3])
>>> conv1 = tf.layers.conv2d(X, filters=96, kernel_size=11, strides=4, padding="SAME")
>>> conv1.shape.as_list()
[None, 56, 56, 96]
>>> conv1 = tf.layers.conv2d(X, filters=96, kernel_size=11, strides=4, padding="VALID")
>>> conv1.shape.as_list()
[None, 54, 54, 96]
```

Here's a little function I use to figure out the feature map sizes:

```python
>>> from math import ceil
>>> def feature_map_size(input_size, kernel_size, stride, padding="SAME"):
...     if padding=="VALID":
...         return int(ceil((input_size - kernel_size + 1) / stride))
...     else:
...         return int(ceil(input_size / stride))
...
>>> feature_map_size(224, 11, 4, padding="VALID")
54
>>> feature_map_size(224, 11, 4, padding="SAME")
56
```

After searching a bit, it seems that this is actually an error in the paper: Andrei Karpathy corrected it in his CS231n Course. Apparently the images were actually 227x227 and the first convolutional layer used VALID padding. Thanks for pointing it out, I'll add a note in my book.

```python
>>> feature_map_size(224, 11, 4, padding="VALID")
57
```

Cheers,
Aurélien

**Comment by fanyike** [comment]
Thanks a lot.
